iphone/CoreApi/CoreApi/Bookmarks/MWMBookmarksManager.mm

```diff
@@ -194,9 +194,18 @@ - (void)loadBookmarks
 
 #pragma mark - Categories
 
-- (BOOL)isCategoryNotEmpty:(MWMMarkGroupID)groupId {
-  return self.bm.HasBmCategory(groupId) &&
-  (self.bm.GetUserMarkIds(groupId).size() + self.bm.GetTrackIds(groupId).size());
+- (BOOL)areAllCategoriesEmpty
```

Comment:
Would it be better/faster/more reusable on Android if implemented in C++?

Reply:
I think yes! I can try to move this logic into the core.

Reply:
I've moved this logic into the cpp.